src/sentry/db/deletion.py

```diff
@@ -15,39 +31,81 @@ def __init__(self, model, project_id=None, dtfield=None, days=None):
         self.days = int(days) if days is not None else None
         self.using = router.db_for_write(model)
 
-    def execute_postgres(self, chunk_size=10000):
-        quote_name = connections[self.using].ops.quote_name
+    def execute_postgres(self, chunk_size=10000, chunk_delta=None):
+        if self.dtfield and self.days is not None:
+            return self.execute_postgres_date(chunk_delta)
 
         where = []
-        if self.dtfield and self.days is not None:
-            where.append("{} < now() - interval '{} days'".format(
-                quote_name(self.dtfield),
-                self.days,
-            ))
         if self.project_id:
-            where.append("project_id = {}".format(self.project_id))
+            where.append("project_id={}".format(self.project_id))
 
         if where:
-            where_clause = 'where {}'.format(' and '.join(where))
+            where_clause = 'WHERE {}'.format(' AND '.join(where))
         else:
             where_clause = ''
 
-        query = """
-            delete from {table}
-            where id = any(array(
-                select id
-                from {table}
-                {where}
-                limit {chunk_size}
-            ));
-        """.format(
+        query = BULK_DELETE_QUERY.format(
             table=self.model._meta.db_table,
             chunk_size=chunk_size,
             where=where_clause,
         )
 
         return self._continuous_query(query)
 
+    def execute_postgres_date(self, chunk_delta=None):
+        if chunk_delta is None:
+            chunk_delta = timedelta(hours=1)
+
+        connection = connections[self.using]
+        dtfield_sql = connection.ops.quote_name(self.dtfield)
```

Comment:
Seems like this should just be called where it's used on line 101 rather than assigned to a local variable and used once ~40 lines later.